### PR TITLE
Script to add and delete Network Virtualisation interfaces

### PR DESCRIPTION
**Pull request by hathyaga**
This script will add and delete vnic interfaces using the
SRIOV capable IO adapter given as an input by the user and
performs this operation multiple times based on the count
given by the user as well

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
job log 
[job.log.txt](https://github.com/avocado-framework-tests/avocado-misc-tests/files/1812193/job.log.txt)

File: io/net/vnic_dlpar.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+#
+#        try:
+#            process.run("startsrc -g rsct_rm", shell=True, sudo=True)
+#        except CmdError as details:
+#            self.log.debug(str(details))
+#            self.fail("Command startsrc -g rsct_rm failed")
+
+        try:
+            for rsct_service in ["startsrc -g rsct", "startsrc -g rsct_rm"]:
+                process.run(rsct_service, shell=True, sudo=True)
```
Comment (narasimhan-v) [inline io/net/vnic_dlpar.py]:
```
for svc in ['rsct', 'rsct_rm']:
    process.run('startsrc -g %s' % rsct_service, shell=True, sudo=True)
```

File: io/net/vnic_dlpar.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Command %s failed", rsct_service)
```
Comment (narasimhan-v) [inline io/net/vnic_dlpar.py]: self.fail("Starting service %s failed", svc)

File: io/net/vnic_dlpar.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+#            process.run("startsrc -g rsct", shell=True, sudo=True)
+#        except CmdError as details:
+#            self.log.debug(str(details))
+#            self.fail("Command startsrc -g rsct failed")
+#
+#        try:
+#            process.run("startsrc -g rsct_rm", shell=True, sudo=True)
+#        except CmdError as details:
+#            self.log.debug(str(details))
+#            self.fail("Command startsrc -g rsct_rm failed")
```
Comment (narasimhan-v) [inline io/net/vnic_dlpar.py]: This can be removed

File: io/net/vnic_dlpar.py.data/vnic_dlpar.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+hmc_username:
+lpar:
+server:
+slot_num:
+vios_name:
+sriov_adapter_id:
+sriov_port:
+bandwidth:
+count:
+
```
Comment (narasimhan-v) [inline io/net/vnic_dlpar.py.data/vnic_dlpar.yaml]: Empty line should be removed.

File: io/net/vnic_dlpar.py [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+            shutil.copy(rpm_install, self.srcdir)
+            os.chdir(self.srcdir)
+            process.run('chmod +x ibmtools')
+            process.run('./ibmtools --install --managed')
+        self.hmc_ip = self.params.get("hmc_ip", '*', default=None)
+        self.hmc_pwd = self.params.get("hmc_pwd", '*', default=None)
+        self.hmc_username = self.params.get("hmc_username", '*', default=None)
+        self.lpar = self.params.get("lpar", '*', default=None)
+        self.server = self.params.get("server", '*', default=None)
+        self.count = self.params.get("count", '*', default=None)
```
Comment (narasimhan-v) [inline io/net/vnic_dlpar.py]: default=1.
Else, the script will error out in the for loop usage.

File: io/net/vnic_dlpar.py [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
```
Comment (abdhaleegit) [inline io/net/vnic_dlpar.py]: Please change it to 2018

File: io/net/vnic_dlpar.py [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+        self.exitcode = exitcode
+
+    def __str__(self):
+        return "Command '%s' exited with %d.\nOutput:\n%s" \
+               % (self.command, self.exitcode, self.output)
+
+
+class Vnic(Test):
+    '''
+    VNIC add/remove script adds a vnic device from an adapter assigned
```
Comment (abdhaleegit) [inline io/net/vnic_dlpar.py]: Please use generic name for above class.. in all places

File: io/net/vnic_dlpar.py.data/README [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+Links :
+-> https://pypi.python.org/packages/source/p/pycrypto/pycrypto-2.6.1.tar.gz#md5=55a61a054aa66812daf5161a0d5d7eda
+
+
+input parameters:
+hmc_ip: ltciofvtr-hmc.pok.stglabs.ibm.com
+hmc_pwd: abc1234
+hmc_username: hscroot
+lpar: ltc-zz1b-lp11_vnic_automation
+server: ltciofvtr-s822l1fsp-SN212CE8A
```
Comment (abdhaleegit) [inline io/net/vnic_dlpar.py.data/README]: Please remove machine informations and credentials

**Comment by hathyaga** [comment]
New PR: https://github.com/avocado-framework-tests/avocado-misc-tests/pull/822

**Pull request closed without merging**